### PR TITLE
[js] Allow SafariDriver to use Safari Technology Preview

### PR DESCRIPTION
**Pull request by mojwang**
- [X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

### CODE REVIEW

**Comment by mojwang** [comment]
@jleyba do you mind taking a look at this? thx in advance...

File: javascript/node/selenium-webdriver/safari.js [hunk 1]
```diff
@@ -149,6 +150,21 @@ class Options {
   }
 
   /**
+   * Instruct the SafariDriver to use the Safari Technology Preview if true.
+   * Otherwise, use the release version of Safari. Defaults to using the release version of Safari.
+   *
+   * @param {Boolean} useTechnologyPreview
```
Comment (jleyba) [inline javascript/node/selenium-webdriver/safari.js]: `boolean`, not `Boolean`

Comment (mojwang) [inline javascript/node/selenium-webdriver/safari.js]: Updated the PR with code review feedback

File: javascript/node/selenium-webdriver/safari.js [hunk 2]
```diff
@@ -200,14 +217,19 @@ class Driver extends webdriver.WebDriver {
...
+    let caps, exe;
+
     if (opt_config instanceof Options) {
       caps = opt_config.toCapabilities();
     } else {
-      caps = opt_config || Capabilities.safari()
+      caps = opt_config || Capabilities.safari();
+    }
+
+    if (caps.get(TECHNOLOGY_PREVIEW_CAPABILITY_KEY)) {
```
Comment (jleyba) [inline javascript/node/selenium-webdriver/safari.js]: This will always be false. You don't set the capability on the top-level capabilities object.

Comment (mojwang) [inline javascript/node/selenium-webdriver/safari.js]: As requested, the capability is not set on the top-level capabilities objects.

**Comment by mojwang** [comment]
@jleyba, does everything look okay or is there any more feedback in case I missed something?

File: javascript/node/selenium-webdriver/safari.js (hunk 2 above)
Comment (jleyba) [inline javascript/node/selenium-webdriver/safari.js]: I meant you need to properly handle extracting options from the capabilities object.
For consistency with Java, you should end up with something alone the lines of

```js
// Some details omitted for brevity

class Options {
  setTechnologyPreview(preview) {
    this.options_['technologyPreview'] = !!preview;
    return this;
  }
}

function useTechnologyPreview(obj) {
    if (obj instanceof Options) {
      return !!obj.options_['technologyPreview']
    }

    if (obj && typeof obj === 'object') {
      return !!obj['technologyPreview'];
    }

    return false;
}


class Driver extends webdriver.WebDriver {
  static createSession(opt_config, opt_flow) {
    let caps, exe;

    if (opt_config instanceof Options) {
      caps = opt_config.toCapabilities();

    } else {
      caps = opt_config || Capabilities.safari();
    }

    if (useTechnologyPreview(caps.get(OPTIONS_CAPABILITY_KEY))) {
      exe = SAFARIDRIVER_TECHNOLOGY_PREVIEW_EXE;
    }
    // ...
  }
}
```

Comment (mojwang) [inline javascript/node/selenium-webdriver/safari.js]: @jleyba, ah gotcha! Pushing up an update in a few minutes...

Comment (mojwang) [inline javascript/node/selenium-webdriver/safari.js]: @jleyba, ah gotcha! Pushing up an update in a few minutes...

File: javascript/node/selenium-webdriver/safari.js [hunk 3]
```diff
@@ -179,6 +196,23 @@ class Options {
   }
 }
 
+/**
+ * @param  {(Options|Object<string, *>)=} o The options object
+ * @return {boolean}
+ */
+function useTechnologyPreview(o) {
+  if (o instanceof Options) {
+    return !!o.options_[TECHNOLOGY_PREVIEW_OPTIONS_KEY];
```
Comment (jleyba) [inline javascript/node/selenium-webdriver/safari.js]: `options_` may be null. Need:

`!!(o.options_ && o.options_[TECHNOLOGY_PREVIEW_OPTIONS_KEY]);`

Comment (mojwang) [inline javascript/node/selenium-webdriver/safari.js]: @jleyba null-check included

**Pull request merged**